src/pages/index.js

```diff
@@ -44,46 +128,85 @@ profileEditBtn.addEventListener("click", () => {
   formValidators["profile-form"].resetValidation();
 });
 
-const handleImageClick = ({ name, link }) => {
-  imageModal.open({ name, link });
+const renderCard = ({ name, link, _id, ownerId, isLiked }, userId) => {
+  const card = new Card({
+    name,
+    link,
+    userId,
+    _id,
+    ownerId,
+    isLiked,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    cardSelector: selectors.cardSelector,
+  });
+  return card.getView();
 };
 
-const renderCard = (cardData) => {
-  const card = new Card(cardData, handleImageClick, selectors.cardSelector);
-  return card.getView();
+const handleImageClick = ({ name, link }) => {
+  imageModal.open({ name, link });
 };
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const card = renderCard(item);
-      section.addItem(card);
-    },
-  },
-  selectors.cardListSelector
+const avatarModal = new PopupWithForm(
+  selectors.avatarModalSelector,
+  (avatarData) => {
+    avatarModal.setButtonText(true);
+    api
+      .editProfilePhoto(avatarData)
+      .then((data) => {
+        userInfo.setAvatar(data.avatar);
+        avatarModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => avatarModal.setButtonText());
+  }
 );
-section.renderItems();
+avatarModal.setEventListeners();
 
 const imageModal = new PopuuWithImage(selectors.imageModalSelector);
 imageModal.setEventListeners();
 
 const cardModal = new PopupWithForm(selectors.cardModalSelector, (cardData) => {
-  const card = renderCard(cardData);
-  section.addItem(card);
-  cardModal.close();
+  cardModal.setButtonText(true);
+  api
+    .createCard(cardData)
+    .then((data) => {
+      section.addItem(renderCard(data, userId));
+      cardModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => cardModal.setButtonText(false));
 });
 cardModal.setEventListeners();
 
 const profileModal = new PopupWithForm(
   selectors.profileModalSelector,
   (profileData) => {
-    userInfo.setUserInfo(profileData.name, profileData.job);
-    profileModal.close();
+    profileModal.setButtonText(true);
+    api
+      .editUserInfo(profileData)
+      .then((profileData) => {
+        userInfo.setUserInfo(profileData.name, profileData.job);
```

Comment:
Needs correction: when I save the profile info the profile element contents disappear